### ISSUE TITLE
error at startup and error at run_this()

### ISSUE THREAD

**Issue by alexzanderr** [jbyuki/one-small-step-for-vimkind#9]
hi.
i've configured the plugin. i've read the readme and the osv.txt in tried to debug a lua file but i cant because of errors.

by running this `:lua require"osv".launch()` or `:lua require"osv".run_this()` you will get
```shell
E5108: Error executing lua ...packer/start/one-small-step-for-vimkind/lua/osv/init.lua:
704: Neovim is waiting for input at startup. Aborting
```

plugin config: `one-small-step-for-vimkind-settings.lua`
```lua
local dap = require "dap"
dap.configurations.lua = {
    {
        type = "nlua",
        request = "attach",
        name = "Attach to running Neovim instance",
--         host = function()
--             local value = vim.fn.input("Host [127.0.0.1]: ")
--             if value ~= "" then return value end
--             return "127.0.0.1"
--         end
-- ,
--         port = function()
--             local val = tonumber(vim.fn.input("Port: "))
--             assert(val, "Please provide a port number")
--             return val
--         end

    }
}

dap.adapters.nlua = function(callback, config)
    callback({ type = "server", host = config.host or "127.0.0.1", port = config.port or 8082})
end
```

packer config:
```lua
	use { "jbyuki/one-small-step-for-vimkind" }
        use { "mfussenegger/nvim-dap" }
```

this is in my init.lua:
```lua
require("one-small-step-for-vimkind-settings")
```

i have many other plugins installed, but i dont think they interfere.

i even put brakepoint in the file with `lua require"dap".toggle_breakpoint()`, didnt work.

**Comment by jbyuki** [comment]
Hi @alexzanderr ,

Thank you for the issue.
The error displayed means that there was an error during the startup of the neovim headless instance.

What does `nvim --headless -c "exit"` display? This should point out to the error in possibly another plugin.

**Comment by alexzanderr** [comment]
![image](https://user-images.githubusercontent.com/54772464/135498647-4b218538-579e-416b-8bc9-10f7ebd9c129.png)

nothing..

**Comment by alexzanderr** [comment]
but i dont understand how should i debug with this plugin.

why i need 2 instances of neovim?

**Comment by jbyuki** [comment]
I ask the question to you. How would you debug something like telescope for example?

If you need the source code displayed with the breakpoint 
 and use the telescope prompt at the same time. Seems difficult with a single instance.

But for the problem, try `nvim --headless`, maybe it would show something.

**Comment by alexzanderr** [comment]
> How would you debug something like telescope for example?
fair enough, for telescope

> Seems difficult with a single instance.
but how vimspector achieved debugging a file (python, javascript, cpp, etc) with a single instance?


out of `nvim --headless`
![image](https://user-images.githubusercontent.com/54772464/135504532-4f3fd6b9-69b3-4f5b-ac96-3c4b963bf1b5.png)

the file line is from my config, seems that i have a problem with `dashboard`.

should i be seeing something related to dap in this output?

**Comment by jbyuki** [comment]
> but how vimspector achieved debugging a file (python, javascript, cpp, etc) with a single instance?

A process is running in the background just like osv is running neovim headless. For cpp it would lldb I guess.

> out of nvim --headless

That's it. It happened in another issue #3 , the best would be to open an issue saying there is a bug in dashboard when running in headless. And give the error location. Seems to be a map() call within dashboard#util#draw_center. 

When the error is fixed, osv should run fine.

**Comment by alexzanderr** [comment]
okey. i will do that.

**Comment by alexzanderr** [comment]
there is a bug in dashboard by running in all modes. i mean, right now, dashboard its not working fine. i will check again my config.

**Comment by jbyuki** [comment]
Good, let me close the issue now. Reopen if it's still not working.

**Comment by jbyuki** [comment]
Only one way to find out.

**Comment by alexzanderr** [comment]
okey, i run `require"osv".run_this()`, didnt happen anything, so i gues its working?

**Comment by jbyuki** [comment]
Probably try to put a breakpoint.

**Comment by alexzanderr** [comment]
i saw this in `htop`
![image](https://user-images.githubusercontent.com/54772464/135512442-0b5b559e-6883-4ffe-9ebb-f85aa1c905ef.png)

**Comment by alexzanderr** [comment]
> Probably try to put a breakpoint.

fair enouh

**Comment by alexzanderr** [comment]
okey i have breakpoint:
![image](https://user-images.githubusercontent.com/54772464/135513012-4d3c960f-4596-4249-a0a9-b9402acc0fce.png)

then i run this
![image](https://user-images.githubusercontent.com/54772464/135513048-1e30388c-c317-4cb2-8746-f3c2094016a3.png)

 the cursor waits a litte bit, then it jumps back in the editor

then what?

**Comment by jbyuki** [comment]
Relax. Try reopening neovim. 

2) open your script
3) change current directory to current file location using `cd %:p:h` for good measues.
4) place your breakpoint
5) call run_this()

Then it should break at line automatically, no need to interact with dap.

**Comment by jbyuki** [comment]
Also what is the rest of the script content? I can only see from line 30.

**Comment by alexzanderr** [comment]
> Also what is the rest of the script content? I can only see from line 30.

empty.

**Comment by alexzanderr** [comment]
okey i did that. it stopped on the breakpoint.

**Comment by jbyuki** [comment]
Then you're good to go. Just read the dap manual. You will find everything you need. It's not facebook chat here.

**Comment by alexzanderr** [comment]
you could link the dap manual ...

**Comment by jbyuki** [comment]
Yeah, i'm on my phone so that's a little bit too much to ask here. Anyway, use my config then.

```lua
local dap = require"dap"
dap.configurations.lua = { 
  { 
    type = 'nlua', 
    request = 'attach',
    name = "Attach to running Neovim instance",
  }
}

dap.adapters.nlua = function(callback, config)
  callback({ type = 'server', host = config.host or "127.0.0.1", port = config.port or 8086 })
end

vim.api.nvim_set_keymap('n', '<F8>', [[:lua require"dap".toggle_breakpoint()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F9>', [[:lua require"dap".continue()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F10>', [[:lua require"dap".step_over()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<S-F10>', [[:lua require"dap".step_into()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F12>', [[:lua require"dap.ui.variables".hover()<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F5>', [[:lua require"osv".launch({port = 8086})<CR>]], { noremap = true })
vim.api.nvim_set_keymap('n', '<F6>', [[:lua require"osv".run_this()<CR>]], { noremap = true })
```
Press F12 to inspect.

**Comment by alexzanderr** [comment]
cool. thanks a lot for everything.

**Comment by jbyuki** [comment]
The dap manual is found in your vim so I'm not sure what to link `:help dap.txt` I guess.

**Comment by jbyuki** [comment]
> cool. thanks a lot for everything.
No worries.

**Comment by jbyuki** [comment]
What about `nvim --headless -c "exit"`? There should be some output because the error precisely says there is an output. It says it starts in blocked mode. ( see `:help nvim_get_mode` for more info )

**Comment by jbyuki** [comment]
Do you have maybe multiple versions of neovim installed? Or are you using some kind of configuration switching mechanism?

**Comment by alexzanderr** [comment]
guys, this is not facebook chat.

**Comment by jbyuki** [comment]
@Melkster Please open a new issue if you still need help.
